lib/plugins/pump.js

```diff
@@ -0,0 +1,265 @@
+        var show = selected || field.level > levels.INFO;
+        if (show) {
+          values.push(field.display);
+        } else {
+          info.push({label: field.label, value: field.display});
+        }
+      }
+    });
+
+    sbx.pluginBase.updatePillText(pump, {
+      value: values.join(' ')
+      , info: info
+      , label: 'Pump'
+      , pillClass: statusClass(result.level)
+    });
+  };
+
+  return pump;
+
+}
+
+function statusClass (level) {
+  var cls = 'current';
+
+  if (level === levels.WARN) {
+    cls = 'warn';
+  } else if (level === levels.URGENT) {
+    cls = 'urgent';
+  }
+
+  return cls;
+}
+
+
+function updateClock (prefs, result, sbx) {
+  if (result.clock) {
+    result.clock.label = 'Last Clock';
+    result.clock.display = timeFormat(result.clock.value, sbx);
+
+    var urgent = moment(sbx.time).subtract(prefs.urgentClock, 'minutes');
+    var warn = moment(sbx.time).subtract(prefs.warnClock, 'minutes');
+
+    if (urgent.isAfter(result.clock.value)) {
+      result.clock.level = levels.URGENT;
+      result.clock.message = 'URGENT: Pump data stale';
+    } else if (warn.isAfter(result.clock.value)) {
+      result.clock.level = levels.WARN;
+      result.clock.message = 'Warning, Pump data stale';
+    } else {
+      result.clock.level = levels.NONE;
+    }
+  }
+}
+
+function updateReservoir (prefs, result) {
+  if (result.reservoir) {
+    result.reservoir.label = 'Reservoir';
+    result.reservoir.display = result.reservoir.value + 'U';
+    if (result.reservoir.value < prefs.urgentRes) {
+      result.reservoir.level = levels.URGENT;
+      result.reservoir.message = 'URGENT: Pump Reservoir Low';
+    } else if (result.reservoir.value < prefs.warnRes) {
+      result.reservoir.level = levels.WARN;
+      result.reservoir.message = 'Warning, Pump Reservoir Low';
+    } else {
+      result.reservoir.level = levels.NONE;
+    }
+  }
+}
+
+function updateBattery (type, prefs, result) {
+  if (result.battery) {
+    result.battery.label = 'Battery';
+    result.battery.display = result.battery.value + type;
+    var urgent = type === 'v' ? prefs.urgentBattV : prefs.urgentBattP;
+    var warn = type === 'v' ? prefs.warnBattV : prefs.warnBattP;
+
+    if (result.battery.value < urgent) {
+      result.battery.level = levels.URGENT;
+      result.battery.message = 'URGENT: Pump Battery Low';
+    } else if (result.battery.value < warn) {
+      result.battery.level = levels.WARN;
+      result.battery.message = 'Warning, Pump Battery Low';
+    } else {
+      result.battery.level = levels.NONE;
+    }
+  }
+}
+
+
+function buildMessage (result) {
+  if (result.level > levels.NONE) {
+    var message = [];
+
+    if (result.battery) {
+      message.push('Pump Battery: ' + result.battery.display);
+    }
+
+    if (result.reservoir) {
+      message.push('Pump Reservoir: ' + result.reservoir.display);
+    }
+
+    result.message = message.join('\n');
+  }
+}
+
+function prepareData (prop, prefs, sbx) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [This function's cyclomatic complexity is too high. (10)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1808382488/issues/source?bid=2452379&fileBranchId=3051223#l201)